central/complianceoperator/v2/checkresults/service/service_impl.go

```diff
@@ -128,8 +131,18 @@ func (s *serviceImpl) GetComplianceProfileStats(ctx context.Context, request *v2
 		return nil, errors.Wrapf(err, "Unable to retrieve compliance profile scan stats for %+v", request)
 	}
 
+	profileMap := map[string]*storage.ComplianceOperatorProfileV2{}
```

Comment:
So we can simplify this one.  If you look at line 110, you can see we require a single profile name as a parameter.  So we don't need to loop through the results searching for the profile.  We know which one they are requesting and we only need to retrieve that one.